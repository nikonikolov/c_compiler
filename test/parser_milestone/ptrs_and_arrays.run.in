int f(){
	while(0){
		a=&b;
		a=*b;
		*ptr++;
		++*ptr;
		b|=a*b-c/g*g%h-b+(*a);
		b|=a*b-c/g*g%h-b+(&a);
		b|=a*b-c/g*g%h-b+&a;
		b|=a*b-c/g*g%h-b+*a;
		*a=b;
		*a=&b;

	}
}

