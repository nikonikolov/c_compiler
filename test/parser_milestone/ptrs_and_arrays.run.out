FUNCTION : f
SCOPE
    WHILE LOOP
    CONDITION: 0
    SCOPE
        (a)=(&(b))
        (a)=(*(b))
        *((ptr)++)
        ++(*(ptr))
        (b)|=(((((a)*(b))-((((c)/(g))*(g))%(h)))-(b))+(*(a)))
        (b)|=(((((a)*(b))-((((c)/(g))*(g))%(h)))-(b))+(&(a)))
        (b)|=(((((a)*(b))-((((c)/(g))*(g))%(h)))-(b))+(&(a)))
        (b)|=(((((a)*(b))-((((c)/(g))*(g))%(h)))-(b))+(*(a)))
        (*(a))=(b)
        (*(a))=(&(b))
